src/main/java/org/opensearch/knn/quantization/factory/QuantizerRegistrar.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.factory;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.opensearch.knn.quantization.enums.ScalarQuantizationType;
+import org.opensearch.knn.quantization.models.quantizationParams.ScalarQuantizationParams;
+import org.opensearch.knn.quantization.quantizer.MultiBitScalarQuantizer;
+import org.opensearch.knn.quantization.quantizer.OneBitScalarQuantizer;
+
+/**
+ * The QuantizerRegistrar class is responsible for registering default quantizers.
+ * This class ensures that the registration happens only once in a thread-safe manner.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+final class QuantizerRegistrar {
+
+    /**
+     * Registers default quantizers if not already registered.
+     * <p>
+     * This method is synchronized to ensure that registration occurs only once,
+     * even in a multi-threaded environment.
+     * </p>
+     */
+    static synchronized void registerDefaultQuantizers() {
```

Comment:
why do we need to synchronized this code block? when the caller of this block is already synchronized? any special reason here?

Reply:
The synchronized  the registerDefaultQuantizers() method adds another layer of thread-safety by ensuring that the registration process itself is atomic. I prefer Redundancy for Safety.